hyracks/hyracks-dataflow-std/src/main/java/org/apache/hyracks/dataflow/std/structures/SerializableVector.java

```diff
@@ -15,37 +41,71 @@ public SerializableVector(IResetableComparableFactory factory, int recordSize){
      * @param frameSize
      * @param recordSize
      */
-    public SerializableVector(IResetableComparableFactory factory, int frameSize, int recordSize){
-
+    public SerializableVector(int recordSize, int frameSize){
+        frames = new ArrayList<>();
+        init(recordSize, frameSize);
     }
 
     @Override
-    public boolean get(int index, IResetableComparable record) {
-        return false;
+    public boolean get(int index, IResetableSerializable record) {
+        if(index >= sVectorSize) {
+            throw new IndexOutOfBoundsException("index out of bound");
+        }
+        int frameIdx = getFrameIdx(index);
+        int offsetInFrame = getOffsetInFrame(index);
+        record.deserialize(frames.get(frameIdx), offsetInFrame, recordSize);
+        return true;
     }
 
     @Override
-    public boolean append(IResetableComparable record) {
-        return false;
+    public boolean append(IResetableSerializable record) {
+        if(sVectorSize % recordPerFrame == 0){    //add a new frame
+            byte[] frame = new byte[frameSize];
+            frameCount ++;
+            record.serialize(frame, 0);
+            frames.add(frame);
+            lastPos = recordSize;
+        }
+        else{
+            int frameIdx = frameCount - 1;
+            int offsetInFrame = lastPos;
+            record.serialize(frames.get(frameIdx), offsetInFrame);
+            lastPos += recordSize;
+        }
+        return true;
     }
 
     @Override
-    public boolean set(int index, IResetableComparable record) {
-        return false;
+    public boolean set(int index, IResetableSerializable record) {
+        if(index >= sVectorSize){
+            throw new IndexOutOfBoundsException("index out of bound");
```

Comment:
If we are throwing the exception, it's better to change the return value from `boolean` to `void`